platform-hub-api/app/models/kubernetes_token.rb

```diff
@@ -0,0 +1,47 @@
+class KubernetesToken
+  include ActiveModel::Model
+  include ActiveModel::Serialization
+  include ActiveModel::Associations
+  include ActiveModel::Validations
+
+  attr_accessor :identity_id, :cluster, :token, :uid, :groups
+
+  validates_presence_of :identity_id, :cluster, :token, :uid, :groups
+  validates_inclusion_of :cluster, in: proc {
+    HashRecord.kubernetes.find_by!(id: 'clusters').data.map {|c| c['id']}
+  }
+  validate :token_must_not_be_blank
+
+  belongs_to :identity
+
+  def initialize(attributes)
+    super(attributes)
+    encrypt_token
+  end
+
+  def [](attr)
+    self.send(attr)
+  end
+
+  def []=(attr, value)
+    self.send("#{attr}=", value)
+  end
+
+  def decrypted_token
+    ENCRYPTOR.decrypt(self.token)
+  end
+
+  protected
+
+  def encrypt_token
+    return unless decrypted_token.nil? # only encrypt if not already encrypted
```

Comment:
What if the value of the token is updated in memory (and thus needs to be encrypted again)? (I know you probably don't use the instance of this class like that anywhere, but it may be possible in the future, so this _could_ be a bug waiting to happen?)